### ISSUE TITLE
Question: hoist-non-react-statics

### ISSUE THREAD

**Issue by bringking** [mobxjs/mobx-react-lite#32]
First thanks for this library, I love the updates compared to the old package. When implementing this in a Next.js codebase, I ran into issues where the `observer` HOC doesn't hoist non-react-statics e.g. getInitialProps. Is this something we should add? If so, I would be happy to make a PR. 

If not, the workaround is easy enough 

```javascript
const ObservedComponent = observer(Component);
ObservedComponent.getInitialProps = () => { };
```

**Comment by bringking** [comment]
@xaviergonz yeah, true `useObserver` might be just as good..

**Comment by FredyC** [comment]
I don't know. Personally, I am no fan of using static properties like this. What's the point? Why it cannot be just a scoped function which you can export along the component if needed? Feels much cleaner in my opinion.

```js
const ObservedComponent = observer(Component);
export const getInitialProps = () => { };
export default ObservedComponent
```

In case you really need it for some weird use case scenario, the `useObserver` could be a better candidate for sure. However, we should probably add some dev-only warning that checks for non-react statics and recommends to `useObserver`. Can you prepare such PR @bringking ?

**Comment by mweststrate** [comment]
Thinking about this, I think we should support hoisting for a few reasons:
1. Some frameworks depend on it, .e.g Next. It can be worked around, but I think finding out that something has not been hoisted is a subtle thing to find in the first thing (in this specific case: a lifecycle is not fired)
2. mobx-react's `observer` currently does supports hoisting for for function components, changing that would be quite a subtle breaking change

So I think there are two ways to fix it:
1. Only do hoisting in `mobx-react`'s observer, making it consistent with `observer` for classes and previous mobx-react versions, but inconsistent with `mobx-react-lite`
2. Do the hoisting in `mobx-react-lite` as a good HoC citizen here, and making the bundle slightly larger by it

**Comment by FredyC** [comment]
Ok, I don't have any experience with Next and how are statics used there. If it's really needed, then I guess we have to obey 👼 

That module feels kinda bulky considering there are just three React related statics for a functional component. It's too much universal imo.

**Comment by mweststrate** [comment]
> That module feels kinda bulky considering there are just three React related statics for a functional component. It's too much universal imo.

It does, it jumps the bundle size from `1.35` to `2.36` KB gzipped. The reason I used it so far is that it takes care of an internal detail about react; the problem is not, how to copy does fields? But more, how to know which fields should _not_ be copied. This might change over time as this is not part of React's public API. But I agree it doesn't feel it justifies almost doubling bundle size. I'll check their sources and see if it is doable to just grab the essential function stuff.

**Comment by FredyC** [comment]
Changes released under version 1.1.0.
